deeppavlov/skills/ecommerce_bot/tfidf_bot.py

```diff
@@ -0,0 +1,275 @@
+# Copyright 2018 Neural Networks and Deep Learning lab, MIPT
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import Counter
+from typing import List, Tuple, Dict, Union
+from operator import itemgetter
+from scipy.stats import entropy
+from scipy.sparse import csr_matrix, vstack
+from scipy.sparse.linalg import norm as sparse_norm
+
+import numpy as np
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.common.log import get_logger
+from deeppavlov.core.common.file import save_pickle, load_pickle
+from deeppavlov.core.commands.utils import expand_path
+from deeppavlov.core.models.estimator import Component
+
+log = get_logger(__name__)
+
+
+@register("ecommerce_tfidf_bot")
+class EcommerceTfidfBot(Component):
+    """Class to retrieve product items from `load_path` catalogs
+    in sorted order according to the similarity measure
+    Retrieve the specification attributes with corresponding values
+    in sorted order according to entropy.
+
+    Parameters:
+        preprocess: text preprocessing component
```

Comment:
?

Reply:
Modern IDE are also able to check consistency between docstrings and signatures.